cmake/DaemonArchitecture.cmake

```diff
@@ -0,0 +1,75 @@
+# Daemon BSD Source Code
+# Copyright (c) 2022, Daemon Developers
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of the <organization> nor the
+#    names of its contributors may be used to endorse or promote products
+#    derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+################################################################################
+# Determine architecture
+################################################################################
+
+# When adding a new architecture, look at all the places ARCH is used
+
+if(DAEMON_DIR)
+	set(TARGET_DETECT_SOURCE_FILE "${DAEMON_DIR}/src/common/Architecture.cpp")
+else()
+	set(TARGET_DETECT_SOURCE_FILE "${CMAKE_CURRENT_SOURCE_DIR}/src/common/Architecture.cpp")
+endif()
+
+try_run(
+	run_result_unused
+	compile_result_unused
+	"${CMAKE_BINARY_DIR}"
+	"${TARGET_DETECT_SOURCE_FILE}"
+	CMAKE_FLAGS CMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES}
+	COMPILE_DEFINITIONS -DPREPROCESSOR_ARCH_DETECTION=1
+	COMPILE_OUTPUT_VARIABLE ARCH
+)
+
+string(REGEX MATCH "DAEMON_ARCH ([a-zA-Z0-9_]+)" ARCH "${ARCH}")
+string(REPLACE "DAEMON_ARCH " "" ARCH "${ARCH}")
+
+if (NOT ARCH)
```

Comment:
I think this condition is not going to work because you recycled the ARCH variable for both the compiler output and the detected architecture